industry_benchmarks/utils/data_gathering.py

```diff
@@ -453,14 +474,267 @@ def gather_ligand_scores(
+        .resolve()
+        .parent
+    )
+    # if the dir doesn't exist, we should try and fix it
+    if not results_dir.is_dir():
+        print("Fixing path to results dir")
+        # Depending on the relative location to the result dir, we might have
+        # to fix a duplicate folder, see this post for more details
+        # https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/83#discussion_r1689003616
+        results_dir = remove_first_reversed_sequential_duplicate_from_path(
+            results_dir
+        )
+        # Now we should check if the dir exists
+        if not results_dir.is_dir():
+            error_message = f"Can't find results directory: {results_dir}"
+            raise FileNotFoundError(error_message)
+
+    # now check that all of the results files can be found in the folder
+    # allow skipping of missing png files
+    for f_name in RESULT_FILES:
+        if not results_dir.joinpath(f_name).exists() and ".png" not in f_name:
+            error_message = f"Can't find cleaned results file: {f_name} in {results_dir}"
+            raise FileNotFoundError(error_message)
+        else:
+            error_message = f"Can't find cleaned results file: {f_name} in {results_dir} skipping"
+            print(error_message)
+
+
+    return results_dir
+
+def get_transform_name(result: dict, alchemical_network: gufe.AlchemicalNetwork) -> str:
+    """
+    Get the name of this transformation, taking into account that the inputs might have accidentally been deleted.
+    """
+    # grab the gufe key of the chemical systems used in the inputs
+    unit_result = list(result["unit_results"].values())[0]
+    state_a_key = unit_result["inputs"]["stateA"][":gufe-key:"]
+    mapping_key = unit_result["inputs"]["ligandmapping"][":gufe-key:"]
+    # work out which system this is in the alchemical network
+    system_look_up = dict((str(node.key), node) for node in alchemical_network.nodes)
+    mapping_look_up = dict((str(edge.mapping.key), edge.mapping) for edge in alchemical_network.edges)
+    # build the transform
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in system_look_up[state_a_key].components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    ligmap = mapping_look_up[mapping_key]
+
+    name = f"{phase}_{ligmap.componentA.name}_{ligmap.componentB.name}"
+    return name
+
+def check_network_is_connected(results_data: dict, alchemical_network: gufe.AlchemicalNetwork) -> bool:
+    """Build a network from the results and check the network is connected."""
+    edges = []
+    for transform in alchemical_network.edges:
+        if transform.name in results_data and len(results_data[transform.name]) == 3:
+            edges.append(transform)
+
+    # extract the ligand network and check its connected
+    result_network = gufe.AlchemicalNetwork(edges=edges)
+    ligand_network = extract_ligand_network(result_network)
```

Comment:
If only one of the legs completed successfully, this would return true, even though the network is technically not connected. We may be checking for both phases to be present elsewhere, but just putting it down here to remember to check that.

Reply:
Great point I added a check specifically for this in [54032b7](https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/179/commits/54032b7f4486f99d6e94d18ee66b9df0efc47dcf)